ruby/ql/src/utils/modeleditor/ModelEditor.qll

```diff
@@ -141,6 +154,17 @@ class MethodEndpoint extends Endpoint instanceof DataFlow::MethodNode {
     or
     not this.isSupported() and result = ""
   }
+
+  /**
+   * Holds if this method is a constructor for a module.
+   */
+  private boolean isConstructor() {
+    if
+      super.getMethodName() = "initialize" and
+      exists(DataFlow::ModuleNode m | m.getOwnInstanceMethod(super.getMethodName()) = this)
+    then result = true
+    else result = false
+  }
```

Comment:
This would be idiomatically a predicate, I think
```ql
private predicate isConstructor() {
  super.getMethodName() = "initialize" and exists(...)
}
```

Then elsewhere you would just write `this.isConstructor()`

Reply:
Thanks! I've updated it.